internal/service/organizations/organizational_unit_descendant_accounts.go

```diff
@@ -0,0 +1,189 @@
+package organizations
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceOrganizationalUnitDescendantAccounts() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceOrganizationalUnitDescendantAccountsRead,
+
+		Schema: map[string]*schema.Schema{
+			"parent_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"accounts": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"email": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"joined_method": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"joined_timestamp": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Todo: Get all child OUs
+func getDescendantOrganizationalUnitsIDs(parent_id string, meta interface{}) ([]string, error) {
+	conn := meta.(*conns.AWSClient).OrganizationsConn
+
+	params := &organizations.ListOrganizationalUnitsForParentInput{
+		ParentId: aws.String(parent_id),
+	}
+
+	// Descendants will hold IDs of all generations of OUs under parent_id.
+	var descendants []string
+	// Children will hold immediate child OUs of parent_id.
+	var children []*organizations.OrganizationalUnit
+	// ChildOUIds will hold the OU IDs of child ous.
+	var childOUIds []string
+	// result will hold any descendants of immediate child OUs.
+	var result []string
+
+	// Get all immediate children OUs.
+	err := conn.ListOrganizationalUnitsForParentPages(params,
+		func(page *organizations.ListOrganizationalUnitsForParentOutput, lastPage bool) bool {
+			children = append(children, page.OrganizationalUnits...)
+
+			return !lastPage
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("error listing Organizations Organization Units for parent (%s): %w", parent_id, err)
+	}
+
+	if childOUIds = getIDsFromOUs(children); childOUIds != nil {
+		for _, id := range childOUIds {
+			// Append this child ou Id and all it's descendants Ids.
+			descendants = append(descendants, id)
+
+			result, err = getDescendantOrganizationalUnitsIDs(id, meta)
+			if err != nil {
+				return descendants, err
```

Comment:
You don't have an `err` handler after this return in the recursion, so the error doesn't halt the function calls.

Reply:
Line 97 is supposed to be the error handler for recursive calls. The initial error is returned on line 87, then all the parent function calls return that error at line 97. What function calls will continue if we return an error here?

Reply:
:face_with_spiral_eyes: yeah, you're right. I messed up on the recursion logic. My bad :sweat_smile: